pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -90,15 +92,29 @@
  */
 @TaskGenerator
 public class MergeRollupTaskGenerator implements PinotTaskGenerator {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MergeRollupTaskGenerator.class);
   private static final int DEFAULT_MAX_NUM_RECORDS_PER_TASK = 50_000_000;
   private static final String REFRESH = "REFRESH";
-  private static final Logger LOGGER = LoggerFactory.getLogger(MergeRollupTaskGenerator.class);
 
+  // This is the metric that keeps track of the task delay in the number of time buckets. For example, if we see this
```

Comment:
Thanks for the detailed comments of the metric.

Reply:
Ah nice, yes.

In that case, does it make sense to make this a time-based metric (e.g. number of hours)? Then it is easier to explain it to admins -- set the limit to the number of hours you can tolerate in terms of delay.

just a suggestion.

Reply:
@mcvsubbu We thought about this approach (initially we wanted to make the delay metric as `n days`). But, we also support multiple merge levels, and this merge level configuration is pretty flexible (it can be 1 day, 30days, or even 1yr). 

For instance, if we are 1 window behind for a 30days window, the time-based metric will quickly go up to 30 days of delay. So, we decided to normalize this delay as `# of windows` so that we can put one global threshold.

Reply:
makes sense, thanks.